pkg/destroy/ovirt/destroyer.go

```diff
@@ -144,6 +148,25 @@ func (uninstaller *ClusterUninstaller) removeTemplate(con *ovirtsdk.Connection)
 	return nil
 }
 
+func (uninstaller *ClusterUninstaller) removeAffinityGroups(con *ovirtsdk.Connection) error {
+	cID := uninstaller.Metadata.Ovirt.ClusterID
+	affinityGroupService := con.SystemService().ClustersService().ClusterService(cID).AffinityGroupsService()
+	res, err := affinityGroupService.List().Send()
+	if err != nil {
+		return err
+	}
+	for _, ag := range res.MustGroups().Slice() {
+		if strings.Contains(ag.MustName(), uninstaller.Metadata.InfraID) {
+			uninstaller.Logger.Infof("Removing AffinityGroup %s", ag.MustName())
+			_, err := affinityGroupService.GroupService(ag.MustId()).Remove().Send()
+			if err != nil {
+				uninstaller.Logger.Errorf("failed to remove AffinityGroup: %s", err)
```

Comment:
Don't we also want to ultimately return an error from `removeAffinityGroups` when we fail to remove an affinity group, so that we keep trying?

But I see that we are not retrying for other resources. I think that is a serious deficiency that needs to be addressed for the ovirt destroyer, in general.

Reply:
Opened a new bug for that https://bugzilla.redhat.com/show_bug.cgi?id=1947424 Thanks!